test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -40,6 +40,31 @@ namespace Npgsql.Tests
 {
     public class ConnectionTests : TestBase
     {
+
+        [OneTimeSetUp]
+        public void CreateTestFile()
+        {
+            // set up pgpass file with connection credentials
+            const string content = "*:*:*:npgsql_tests:npgsql_tests";
+            var pgpassFile = PgPassFile.GetPgPassFilePath();
+            var directory = Path.GetDirectoryName(pgpassFile);
+            if (directory != null)
+            {
+                Directory.CreateDirectory(directory);
+            }
+            File.WriteAllText(pgpassFile, content);
+        }
+
+        [OneTimeTearDown]
+        public void DeleteTestFile()
+        {
+            var pgpassFile = PgPassFile.GetPgPassFilePath();
+            if (File.Exists(pgpassFile))
+            {
+                File.Delete(pgpassFile);
```

Comment:
Umm, this deletes the actual pgpass.conf file on the system, even if it existed regardless of unit tests... It's very problematic for unit tests to change the environment in this way.

IMHO for this feature it could be enough to test manually. To make a less invasive unit test you could copy any current pgpass aside and restore it at the end of the tests.

Another problem with the code is that you're hard-coding npgsql_tests as the user and password, but the Npgsql testsuite allows you to change the connection string via an environment variable. You should extract the username and password from `TestBase.ConnectionString`.